### PR TITLE
Add references for R2 and LBFGS in readme #236

### LINKED ISSUE #236: Add references for R2 and LBFGS in readme

### PR DESCRIPTION
**Pull request by hpmouton**
This pull request addresses issue #236 by adding references for the R2 and LBFGS implementations in the project's readme. 

Changes Made:

* Added references for the R2 implementation in the readme.
* Included references for the LBFGS implementation in the readme.

Related Issue:
Closes #236

### CODE REVIEW

**Comment by tmigot** [comment]
Thanks @hpmouton for the suggestion! I really like the idea.

@dpo or @d-monnet Would you have some time to check if that's the "right" references?

**Comment by d-monnet** [comment]
> Thanks @hpmouton for the suggestion! I really like the idea.
> 
> @dpo or @d-monnet Would you have some time to check if that's the "right" references?

As far as I know, there's no reference for R2 itself. I'm not sure our paper with D. Orban is the most relevant reference, I'll let @dpo decide what reference is the most appropriate.

File: README.md [hunk 1]
```diff
@@ -21,7 +21,15 @@ and bound-constrained optimization
 This package provides an implementation of four classic algorithms for unconstrained/bound-constrained nonlinear optimization:
 
 - `lbfgs`: an implementation of a limited-memory BFGS line-search method for unconstrained minimization;
+    > Boggs, P. T., & Byrd, R. H. (2019). Adaptive, Limited-Memory BFGS
+    > Algorithms for Unconstrained Optimization. SIAM Journal on
+    > Optimization, 29(2), 1282-1299.
+    > DOI: [10.1137/16M1065100](https://doi.org/10.1137/16M1065100)
```
Comment (dpo) [inline README.md:27]: We don't implement that strategy. Our implementation is close to that of https://link.springer.com/article/10.1007/BF01589116 but with a different line search.

File: README.md [hunk 2]
```diff
@@ -21,7 +21,15 @@ and bound-constrained optimization
...
+    
 - `R2`: a first-order quadratic regularization method for unconstrained optimization;
+  > Monnet, Dominique & Orban, Dominique. (2023). A Multi-Precision Quadratic
+  > Regularization Method for Unconstrained Optimization with Rounding Error Analysis.
+  > DOI: [10.13140/RG.2.2.22068.01926.](http://dx.doi.org/10.13140/RG.2.2.22068.01926)
```
Comment (dpo) [inline README.md:32]: The version of R2 here is not the one described in that paper. One reference for R2 is https://link.springer.com/article/10.1007/s10107-016-1065-8 (R2 is a special case).

**Review by dpo — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (hpmouton) [reply README.md]: I added the citation for the suggested reference.

File: README.md [hunk 3]
```diff
@@ -21,15 +21,18 @@ and bound-constrained optimization
 This package provides an implementation of four classic algorithms for unconstrained/bound-constrained nonlinear optimization:
 
 - `lbfgs`: an implementation of a limited-memory BFGS line-search method for unconstrained minimization;
-    > Boggs, P. T., & Byrd, R. H. (2019). Adaptive, Limited-Memory BFGS
-    > Algorithms for Unconstrained Optimization. SIAM Journal on
-    > Optimization, 29(2), 1282-1299.
-    > DOI: [10.1137/16M1065100](https://doi.org/10.1137/16M1065100)
+    > D. C. Liu, J. Nocedal. (1989). On the limited memory BFGS method for
+    > large scale optimization. *Mathematical Programming*, 45(1), 503-528.
+    > DOI: [10.1007/BF01589116](https://doi.org/10.1007/BF01589116)
```
Comment (dpo) [inline README.md:26]: It seems like the indentation of the reference here is not the same as in the other references.

**Review by dpo — CHANGES_REQUESTED** [review]
Thank you! Just a minor final comment.

File: README.md [hunk 4]
```diff
@@ -21,7 +21,18 @@ and bound-constrained optimization
 This package provides an implementation of four classic algorithms for unconstrained/bound-constrained nonlinear optimization:
 
 - `lbfgs`: an implementation of a limited-memory BFGS line-search method for unconstrained minimization;
+    > D. C. Liu, J. Nocedal. (1989). On the limited memory BFGS method for
```
Comment (dpo) [inline README.md:24]:
```suggestion

    > D. C. Liu, J. Nocedal. (1989). On the limited memory BFGS method for
```

File: README.md [hunk 5]
```diff
@@ -21,7 +21,18 @@ and bound-constrained optimization
...
+    > large scale optimization. *Mathematical Programming*, 45(1), 503-528.
+    > DOI: [10.1007/BF01589116](https://doi.org/10.1007/BF01589116)
+
+    
 - `R2`: a first-order quadratic regularization method for unconstrained optimization;
+    > E. G. Birgin, J. L. Gardenghi, J. M. Martínez, S. A. Santos, Ph. L. Toint. (2017).
```
Comment (dpo) [inline README.md:30]:
```suggestion

    > E. G. Birgin, J. L. Gardenghi, J. M. Martínez, S. A. Santos, Ph. L. Toint. (2017).
```

**Review by dpo — CHANGES_REQUESTED** [review]

**Review by dpo — APPROVED** [review]

**Comment by dpo** [comment]
Thank you!

**Pull request merged**